bin/reference_data.py

```diff
@@ -208,18 +222,33 @@ def upload_to_rest_api(self):
             genome_size = input(
                 "Enter species genome size (in Mb) to use for yield calculation. (default: %.0f) " % genome_size
             ) or genome_size
-            # FIXME: Probably should expose the taxid in EGCG-Core so we do not have to access the private method
-            q, taxid, scientific_name, common_name = ncbi._fetch_from_cache(self.species)
+            # FIXME: Probably should expose the taxid in EGCG-Core so we do not have to access the private methods
+            info = ncbi._fetch_from_cache(self.species)
+            if info:
+                q, taxid, scientific_name, common_name = info
+            else:
+                taxid, scientific_name, common_name = ncbi._fetch_from_eutils(self.species)
+
             rest_communication.post_entry(
                 'species',
                 {
                     'name': self.species,
                     'genomes': [self.genome_version],
                     'default_version': self.genome_version,
                     'taxid': taxid,
-                    'approximate_genome_size': genome_size
+                    'approximate_genome_size': float(genome_size)
                 }
             )
+    @staticmethod
+    def genome_param_from_fai_file(fai_file):
+        genome_size = 0
+        nb_chromosome = 0
+        with open(fai_file) as open_file:
+            for line in open_file:
+                sp_line = line.strip().split()
```

Comment:
A comment explaining the following lines would be helpful.

Reply:
Added docstring to the function

Reply:
Cheers